Add `README` to `bullet_train-api` gem

newstler opened the pull request:
Here's my attempt to write the `README` for `bullet_train-api` gem.

gazayas commented on bullet_train-api/README.md:
```diff
@@ -1,28 +1,131 @@
+
+#### Create separate translations for API
+
+Starting 1.6.27, Bullet Train Scaffolding generates separate translations for API documentation: `api_title` and `api_description`.
```
"Starting in 1.6.27"

gazayas commented on bullet_train-api/README.md:
```diff
@@ -1,28 +1,131 @@
+which uses FactoryBot to build an in-memory representation of the model,
+then generates the relevant OpenAPI schema for that model.
+
+ExampleBot will attempt to create a instance of the given model called `<model>_example`.
```
"an instance"

gazayas reviewed (commented):
Thanks for writing this up! I just wrote a couple of comments to fix in the README.

newstler commented:
@gazayas Thanks! That's still work in progress, I plan to slowly gather all the information possible in next 1-2 weeks. Feel free to add anything you think worth mentioning in README!

jagthedrummer commented:
@newstler, just wondering if we should go ahead and merge this. In my opinion publishing documentation as it's in progress is better than keeping it hidden until it's 100% perfect. There seems to be a _lot_ of good info here that's otherwise completely unavailable to people. I'd love to get it out as soon as we can, even if we need to tweak it again later.

newstler commented:
Hey @jagthedrummer,

I totally agree with you. Unfortunately didn't have time to finish that. Let me run my eyes again quickly on what could be fixed fast and let's merge it.

newstler commented:
@jagthedrummer Just looked at it, feels pretty solid without noticeable dangling phrases, and I don't really remember what I wanted to add more. Rebased it over `main` and added a small detail. From me it's ready to be merged.
